presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/GatherAndMergeWindows.java

```diff
@@ -176,10 +170,21 @@ public Result apply(WindowNode parent, Captures captures, Context context)
         }
     }
 
-    private static Set<VariableReferenceExpression> extractUnique(Assignments assignments, TypeProvider types)
+    private static Set<VariableReferenceExpression> extractUnique(Assignments assignments)
     {
         Collection<RowExpression> expressions = assignments.getExpressions();
-        return VariablesExtractor.extractUnique(expressions.stream().map(OriginalExpressionUtils::castToExpression).collect(toImmutableList()), types);
+        return VariablesExtractor.extractUnique(expressions);
+    }
+
+    private static boolean dependsOn(WindowNode parent, WindowNode child)
+    {
+        return parent.getPartitionBy().stream().anyMatch(child.getCreatedVariable()::contains)
```

Comment:
This return statement is very hard to read...Though I understand it is a copy-paste, maybe consider using variable name to represent each disjunctions?

Reply:
This is a bit complicated.... I'm not sure if I understand the logic here correctly as well.

Reply:
Let's keep it as is then :(

Reply:
Ya, a clearer way of writing it might be to extract all variables of parent into a collection, and then call `anyMatch(child.getCreatedVariable()::contains)` once. But I don't have strong opinions.